src/battle_ai_main.c

```diff
@@ -1699,6 +1699,7 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                 ADJUST_SCORE(-6);
             break;
         case EFFECT_BELLY_DRUM:
+        case EFFECT_FILLET_AWAY:
```

Comment:
Both Belly Drum and Fillet Away are missing logic for when the stats are already maxed